gsadjust/pyqt_models.py

```diff
@@ -2067,30 +2071,26 @@ def data(self, index, role):
                         brush = QtGui.QBrush(QtCore.Qt.red)
                 return brush
             if role == QtCore.Qt.DisplayRole:
-                if column == DELTA_STATION1:
-                    return delta.sta1
-                elif column == DELTA_STATION2:
-                    return delta.sta2
-                elif column == LOOP:
+                def delta_station_loop():
                     if delta.loop is None:
                         if type(delta.station2) == list:
                             return delta.station2[0].loop
                         else:
                             return "NA"
                     else:
                         return delta.loop
-                elif column == DELTA_TIME:
-                    return dt.datetime.utcfromtimestamp((delta.time() - 719163) * 86400.).strftime('%Y-%m-%d %H:%M:%S')
-                elif column == DELTA_G:
-                    return QVariant("%0.1f" % delta.dg)
-                elif column == DELTA_DRIFTCORR:
-                    return QVariant("%0.1f" % delta.driftcorr)
-                elif column == DELTA_SD:
-                    return QVariant("%0.1f" % delta.sd)
-                elif column == DELTA_ADJ_SD:
-                    return QVariant("%0.1f" % delta.sd_for_adjustment)
-                elif column == DELTA_RESIDUAL:
-                    return QVariant("%0.1f" % delta.residual)
+
+                return {
+                    DELTA_STATION1: delta.sta1,
+                    DELTA_STATION2: delta.sta2,
+                    LOOP: delta_station_loop(),
+                    DELTA_TIME:  dt.datetime.utcfromtimestamp((delta.time() - 719163) * 86400.).strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
This and the delta_station_loop start to show the limits of this pre-calculate approach vs. storing these as functions to lookup the data.